src/main/java/bio/terra/cli/serialization/userfacing/UFClonedResource.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.cli.serialization.userfacing;
+
+import bio.terra.cli.utils.UserIO;
+import bio.terra.workspace.model.CloneResourceResult;
+import bio.terra.workspace.model.ResourceCloneDetails;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.io.PrintStream;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringEscapeUtils;
+
+/**
+ * User-facing view of cloned resource descriptions.
+ *
+ * <p>This is a POJO class intended for serialization. This JSON format is user-facing.
+ */
+@JsonDeserialize(builder = UFClonedResource.Builder.class)
+public class UFClonedResource {
+  public final CloneResourceResult result;
+  public final UFResource sourceResource;
+  @Nullable public final UFResource destinationResource;
+  @Nullable public final String errorMessage;
+
+  public UFClonedResource(
+      ResourceCloneDetails resourceCloneDetails,
+      UFResource sourceResource,
+      @Nullable UFResource destinationResource) {
+    this.result = resourceCloneDetails.getResult();
+    this.sourceResource = sourceResource;
+    this.destinationResource = destinationResource;
+    this.errorMessage = resourceCloneDetails.getErrorMessage();
+  }
+
+  protected UFClonedResource(Builder builder) {
+    this.result = builder.result;
+    this.sourceResource = builder.sourceResource;
+    this.destinationResource = builder.destinationResource;
+    this.errorMessage = builder.errorMessage;
+  }
+
+  public void print() {
+    final String indent = "    ";
+    PrintStream OUT = UserIO.getOut();
+    OUT.println("Source resource:");
+    sourceResource.print(indent);
+    OUT.println();
+    if (null != destinationResource) {
+      OUT.println("Destination resource:");
+      // TODO: just print name, description, and cloud details for destination
```

Comment:
Comment still relevant?

Reply:
I could still do that, but it's ugly. I'd need to either downcast the resource or put in another abstract method for just printing the resoruce-type-specific bits needed here.